data/org.freedesktop.portal.SaveRestore.xml

```diff
@@ -0,0 +1,177 @@
+      via integration with their UI toolkit.
+
+      This documentation describes version 1 of this interface.
+  -->
+  <interface name="org.freedesktop.portal.SaveRestore">
+    <!--
+        Register:
+        @options: Vardict with optional further information
+        @session_handle: Object path for the created :ref:`org.freedesktop.portal.Session` object
+        @registration: Vardict directing the app how to restore its state
+
+        Registers this app with the save/restore API. If the user logs out while
+        the app is registered, the system will remember that this app was running
+        and can relaunch it on next log-in. The app can then register itself again
+        on next startup to be notified of what state (if any) it should restore.
+
+        During a clean shutdown, the app should close the created session using
+        :ref:`org.freedesktop.portal.Session.Close`. If the app disconnects from
+        the bus without closing the session first, the system will consider the
+        shutdown unclean and will report this to the app on its next startup.
+
+        Supported keys in the @options vardict include:
+
+        * ``session_handle_token`` (``s``)
+
+          A string that will be used as the last element of the session handle. Must be a valid
+          object path element. See the :ref:`org.freedesktop.portal.Session` documentation for
+          more information about the session handle.
+
+        Supported keys in the @registration vardict include:
+
+        * ``instance-id`` (``s``)
+
+          A short alphanumeric string that identifies different instances of a
+          multi-instance app. This allows multi-instance apps to keep the state
+          of each instance separated. Single-instance apps should use this as
+          well, because the system will return different identifiers when running
+          different graphical sessions. For instance: if a user switches between
+          GNOME and KDE regularly, even single-instance apps will need to save
+          and load their state under different instance IDs.
+
+        * ``restore-reason`` (``u``)
+
+          Tells the app which state it needs to restore. Supported values:
+
+          - ``PRISTINE``: 0. The app should not attempt to restore any state.
+
+          - ``LAUNCH``: 1. This is a normal launch, so the app should restore
+            some state (i.e. previous window size and position, default view on
+            the home page) but it shouldn't attempt a full state restore. The
+            exception is apps that have a preference that allows the user to
+            decide if the app should always save and restore its state: if the
+            user opts into this behavior, then it's permissible for the app to
+            treat ``LAUNCH`` as equivalent to ``RESTORE``.
+
+          - ``RECOVER``: 2. This is the first launch after the app unexpectedly
```

Comment:
Split this into two different reasons: recover after compositor crash, and recover after app crash